src/main/java/net/neoforged/neoforge/network/payload/AdvancedContainerSetDataPayload.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.network.payload;
+
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.network.protocol.common.custom.CustomPacketPayload;
+import net.minecraft.network.protocol.game.ClientboundContainerSetDataPacket;
+import net.minecraft.resources.ResourceLocation;
+import net.neoforged.neoforge.internal.versions.neoforge.NeoForgeVersion;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * A custom payload that updates the full dataslot value instead of just the short value
+ *
+ * @param containerId The containerId for the container.
+ * @param dataId      The ID of the dataslot.
+ * @param value       The value of the dataslot.
+ */
+@ApiStatus.Internal
+public record AdvancedContainerSetDataPayload(int containerId, int dataId, int value) implements CustomPacketPayload {
+
+    public static final ResourceLocation ID = new ResourceLocation(NeoForgeVersion.MOD_ID, "advanced_set_data");
```

Comment:
```suggestion
    public static final ResourceLocation ID = new ResourceLocation(NeoForgeVersion.MOD_ID, "advanced_set_container_data");
```

Reply:
I'm fine with adding the container string, but shouldn't it then be `advanced_container_set_data` to match the class name?

Reply:
Sure, just add container.